### PR TITLE
Enable Volatile Layer Client write tests

### PR DESCRIPTION
**Pull request by serhiikobalt**
* fix disabled tests
 * refactor cancellation tests

Relates-To: OLPEDGE-710

Signed-off-by: Serhii Lysenko <ext-serhii.lysenko@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp [hunk 1]
```diff
@@ -1923,44 +1972,48 @@ TEST_P(StreamLayerClientCacheMockTest, FlushDataMultiple) {
...
-  auto response = cancel_future.GetFuture().get();
+  auto promise = client_->Flush();
+  waitForCancel->get_future().get();
+  promise.GetCancellationToken().cancel();
+  pauseForCancel->set_value();
+
+  auto response = promise.GetFuture().get();
 
   ASSERT_EQ(1, response.size());
+  testing::Mock::VerifyAndClearExpectations(network_.get());
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: Adding additionally this is not really necessary as google does it for you https://github.com/google/googletest/blob/master/googlemock/docs/cheat_sheet.md#verifying-and-resetting-a-mock
-> *gMock will verify the expectations on a mock object when it is destructed ...*

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp [hunk 2]
```diff
@@ -112,6 +114,17 @@ void PublishSdiiSuccessAssertions(
   EXPECT_FALSE(result.GetResult().GetTraceID().GetGeneratedIDs().at(0).empty());
 }
 
+template <typename T>
+void PublishCancelledAssertions(
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: What is PublishCancelledAssertions?
Do you mean CheckResponse?

Comment (serhiikobalt) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: These assertions are shared across all write tests where publish requests are cancelled.

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 1 above)
Comment (serhiikobalt) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: sometimes network mock might be never destroyed as reference to is is hold by expectation inside the mock

Comment (andescu) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: Then move it to TearDown() please

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 2 above)
Comment (andescu) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: The name of the function is wrong. What do you publish? I dont see any publish? What you do in the function is checking if the result is ok. So the function should be called CheckResponse()

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 1 above)
Comment (serhiikobalt) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: In my opinion TearDown is not the place to verify assertions or expectations on mocks event.

File: olp-cpp-sdk-dataservice-write/test/unit/TestIndexLayerClient.cpp [hunk 3]
```diff
@@ -115,6 +117,14 @@ class IndexLayerClientTestBase : public ::testing::TestWithParam<bool> {
     return index;
   }
 
+  std::string PrintError(const olp::client::ApiError& error) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-write/test/unit/TestIndexLayerClient.cpp]: This is not used. Remove?

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 2 above)
Comment (serhiikobalt) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: there are already functions named 
```
PublishDataSuccessAssertions()
PublishFailureAssertions()
```
Also similar names are used in tests for other layer types. I agree the naming is wrong but at least it is consistent.

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 1 above)
Comment (andescu) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: En contraire mon frere, see https://github.com/google/googletest/blob/master/googletest/docs/faq.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-setupteardown-ctorvssetup

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 2 above)
Comment (andescu) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: When moving you have the chance to change it.

File: olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp (hunk 1 above)
Comment (serhiikobalt) [reply olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp]: I didn't get that. It tells that I might want to prefer TearDown over destructors. Nothing there tells that VerifyAndClearExpectations should be preferably called in TearDown and there are plenty of examples when it called in tests. Placing this call inside TearDown might help code deduplication, but believe having all assertions in one place at the buttom of each test is more readable and less astonishing.

**Pull request merged**